### PR TITLE
feature: sql mi compatibility

### PR DESCRIPTION
**Pull request by Joey40**
## Description
Remove the default option for xpDirTree to allow for compatibility with Azure SQL MI

## How Has This Been Tested?
Not tested

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue) - Fixes #
- [ ] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have read/followed the [**CONTRIBUTING**](https://github.com/LowlyDBA/lowlydba.sqlserver/blob/main/CONTRIBUTING.md) document.
- [x] I have read/followed the [PR Quick Start Guide](https://docs.ansible.com/ansible/devel/community/create_pr_quick_start.html)
- [x] I have added tests to cover my changes or they are N/A.
- [ ] New module options/parameters include a [`version_added` property](https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-fields).

### CODE REVIEW

**Comment by Joey40** [comment]
PR to address #140

**Comment by lowlydba** [comment]
Github Actions are sort of having a day, but a few things also needed:

> ERROR: Found 1 validate-modules issue(s) which need to be resolved:
>ERROR: plugins/modules/restore.ps1:0:0: doc-default-does-not-match-spec: Argument 'xp_dirtree' in argument_spec defines > default as (None) but documentation defines default as (False)

Referring to updating the doc definitions in [restore.py](https://github.com/lowlydba/lowlydba.sqlserver/blob/main/plugins/modules/restore.py).

Can you also add a changelog fragment so that this change gets wrapped up in the ansible release process (example [here](https://github.com/lowlydba/lowlydba.sqlserver/pull/96/files#diff-657c19616f2641d73097fd95095170a5b65a2b73d888882d54a4d21dada46595) - file is deleted once it gets added to the changelog so ignore the fact that it is deleted)

**Review by lowlydba — CHANGES_REQUESTED** [review]
See above comment

File: changelogs/fragments/xpdirtree_sql_mi_bugfix.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- bugfixes:
+  - Removed the default value for xp_dirtree to allow compatibility with Azure SQL Mangaed instances
```
Comment (lowlydba) [inline changelogs/fragments/xpdirtree_sql_mi_bugfix.yml:2]:
```suggestion
bugfixes:
  - Removed the default value for xp_dirtree to allow compatibility with Azure SQL Mangaed instances
```

**Comment by Joey40** [comment]
I made those changes and incorporated your suggestion.

**Comment by lowlydba** [comment]
Thanks! Going to try to wrangle as many green checks as I can since I'm not sure what is causing these failures, but should be all set as far as your changes are concerned.

**Review by lowlydba — APPROVED** [review]

**Comment by lowlydba** [comment]
Issue with the CI is addressed here - https://github.com/ansible-collections/news-for-maintainers/issues/28

Will make a commit to see if that resolves the issues.

**Comment by lowlydba** [comment]
@Joey40 If you update your fork from main we should be able to get a green build now 🎉

**Comment by Joey40** [comment]
done

**Comment by Joey40** [comment]
Awesome, I'll test once it's published.  I imagine there will be other issues to work out for SQL MI support.

**Pull request merged**